app/models/doi/graphql_query.rb

```diff
@@ -0,0 +1,355 @@
+# frozen_string_literal: true
+
+module Doi::GraphqlQuery
+  class Builder
+    include Modelable
+
+    DEFAULT_CURSOR = [0, ""]
+    DEFAULT_PAGE_SIZE = 0
+    DEFAULT_FACET_COUNT = 10
+
+    def initialize(query, options)
+      @query = query
+      @options = options
+    end
+
+    def search_query
```

Comment:
Maybe explicitly name t his like: build_full_search_query
This possibly then gives the notion of building a query without everything listed.